cumulus/client/consensus/aura/src/collators/slot_based/block_builder_task.rs

```diff
@@ -334,13 +332,21 @@ where
 						slot = ?para_slot.slot,
 						"Not building block."
 					);
-					if let Some(ref mut connection_helper) = maybe_connection_helper {
-						connection_helper.update::<Block, P>(para_slot.slot, parent_hash).await;
+
+					if let Ok(authorities) = para_client.runtime_api().authorities(parent_hash) {
+						connection_helper.update::<P>(para_slot.slot, &authorities).await;
 					}
```

Comment:
Same here. Instead of having this twice, just have it once above. Internally we are any way checking if we should do something or not.